### PR TITLE
added 2 buttons to move agendaitems up and down in the list

### PR DESCRIPTION
**Pull request by StuyckRuben**

### CODE REVIEW

File: app/components/agenda/agenda-overview.hbs [hunk 1]
```diff
@@ -77,6 +77,9 @@
                   @showFormallyOkStatus={{@currentAgenda.status.isDesignAgenda}}
                   @isEditingFormallyOkStatus={{@isEditingOverview}}
                   @showDragHandle={{this.canDragAgendaitems}}
+                  @onMove={{perform this.move}}
+                  @isFirst={{and (eq index 0) (eq indexItems 0)}}
+                  @isLast={{and (not (has-next agendaitemGroup @notaGroups)) (not (has-next agendaitem agendaitemGroup.agendaitems ))}}
```
Comment (sergiofenoll) [inline app/components/agenda/agenda-overview.hbs]: Whitespace

```suggestion
                  @isLast={{and (not (has-next agendaitemGroup @notaGroups)) (not (has-next agendaitem agendaitemGroup.agendaitems))}}
```

File: app/components/agenda/agenda-overview.js [hunk 2]
```diff
@@ -26,4 +27,34 @@ export default class AgendaOverview extends Component {
   get canDragAgendaitems() {
     return this.canEdit && this.args.isEditingOverview;
   }
+
+  @task
```
Comment (sergiofenoll) [inline app/components/agenda/agenda-overview.js]: Why is this a `task` instead of an `action`?

File: app/components/agenda/agenda-overview.js [hunk 3]
```diff
@@ -26,4 +27,34 @@ export default class AgendaOverview extends Component {
...
+    let itemArray = [];
+    if (changedAgendaItemType.uri === CONSTANTS.AGENDA_ITEM_TYPES.NOTA) {
+      for (const notaGroup of this.args.notaGroups) {
+        if (notaGroup.agendaitems) {
+          for (const nota of notaGroup.agendaitems) {
+            itemArray.push(nota);
+          }
+        }
+      }
+    } else if ( changedAgendaItemType.uri === CONSTANTS.AGENDA_ITEM_TYPES.ANNOUNCEMENT) {
```
Comment (sergiofenoll) [inline app/components/agenda/agenda-overview.js]: Whitespace

```suggestion
    } else if (changedAgendaItemType.uri === CONSTANTS.AGENDA_ITEM_TYPES.ANNOUNCEMENT) {
```

File: app/components/agenda/agenda-overview.js [hunk 4]
```diff
@@ -26,4 +27,34 @@ export default class AgendaOverview extends Component {
...
+      itemArray = [...this.args.announcements];
+    }
+    for (let i = 0; i < itemArray.length; i++) {
+      if (itemArray[i].id === agendaitem.id) {
+        agendaitemIndex = i;
+        break;
+      }
+    }
+    if(agendaitemIndex > -1 &&
```
Comment (sergiofenoll) [inline app/components/agenda/agenda-overview.js]: Whitespace

```suggestion
    if (agendaitemIndex > -1 && 
```

**Comment by ValenberghsSven** [comment]
Wanted to approve this, but during test I found out you can use the feature in combination with "toon aangepaste punten".
I don't think we want this?
It has very strange side effects:
ex: 1 and 4 are changed. Moved 1 down > 4 became 1 and 1 became 2
When using "toon alle punten" I got 1, 2, 2, 3

I'm guessing this problem also exists for the current dragging feature..

Perhaps we leave it as-is but change the text above?
image

Add something like "Opgelet!. Slepen of verplaatsen van agendapunten in combinatie met "toon aangepaste punten" kan onverwachte resultaten geven." ?

**Comment by tdn** [comment]
Oh I think we definitely want to disable this when the 'toon aangepaste agendapunten' is toggled! Didn't think it would be enabled even with the current dragging function 

@StuyckRuben can you add this to the PR please? Thanks!

**Review by ValenberghsSven — APPROVED** [review]
Changed it so that the drag or move buttons are not available when we are showing the modified items to avoid items being numbered incorrectly.

**Pull request merged**